src/Rules/Cast/CastObjectToStringRule.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Cast;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\ObjectWithoutClassType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeTraverser;
+use PHPStan\Type\UnionType;
+use PHPStan\Type\VerbosityLevel;
+use function sprintf;
+
+/**
+ * @implements Rule<Node\Expr\Cast>
+ */
+class CastObjectToStringRule implements Rule
+{
+
+	public function __construct()
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Node\Expr\Cast::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node instanceof Node\Expr\Cast\String_) {
+			return [];
+		}
+
+		$type = $scope->getType($node->expr);
+		if ($type->toString() instanceof ErrorType) {
+			return [];
+		}
+
+		$objectType = TypeTraverser::map($type, static function (Type $type, callable $traverse) {
+			if ($type instanceof UnionType) {
+				return $traverse($type);
+			}
+
+			if (!(new ObjectWithoutClassType())->isSuperTypeOf($type)->yes()) {
+				return new NeverType();
+			}
+
+			return $type;
+		});
+
+		if ($objectType->hasMethod('__toString')->maybe()) {
```

Comment:
Wouldn't it better to write
```suggestion
		if ($this->reportMaybe && $objectType->hasMethod('__toString')->maybe()) {
```
And add the rule on every level ?

This way if someone is level x but use 
```
reportMaybe: true
```
it will be reported.

And on the opposite side, using
```
reportMaybe: false
```
disable the rule.

Reply:
The rule is on level 7. reportMaybes will never be false.

Reply:
> The rule is on level 7. reportMaybes will never be false.

You can override the value in the config.

I already have/had project with config
```
parameters:
    level: 6
    reportMaybe: true
```
or 
```
parameters:
    level: 7
    reportMaybe: false
```

This is useful when bumping level one by one.
It's a smaller step than bumping from 6 to 7 which is doing
```
parameters:
	checkUnionTypes: true
	reportMaybes: true
	checkListType: true
```
I can do one PR 6 + checkList, then later another PR with 6 + checklist and reportMaybe then a last PR with. 6 + everything, and then change the config to the level 7.

I feel like it's easier to opt-in, opt-out the rules when they are config dependent rather than level dependent. But that's a small difference.

Reply:
You're relying on internal config parameters that aren't documented on phpstan.org. This isn't a supported code path.

Reply:
> You're relying on internal config parameters that aren't documented on phpstan.org. This isn't a supported code path.

Oh, I never thought there was internal config parameter. I see, I understand then.

Reply:
But then, the following example https://phpstan.org/r/658bbf1b-b6d6-4b8e-a844-449c42cb75ed
should be reported as an error on a lower level than the level 7

On PHP level 8 the check 
```
$objectType->hasMethod('__toString')->no()
```
should be true. (I don't know if it does currently)

So it would make sens to have this rule on a lower level with the following check
```
if (
    $objectType->hasMethod('__toString')->no()
    || $this->reportMaybe && $objectType->hasMethod('__toString')->maybe()
) {
```
no ?